src/components/NewGame/hooks/useNewGame.ts

```diff
@@ -0,0 +1,317 @@
+import { useReducer, useCallback } from 'react';
+import Filter from 'bad-words';
+
+import { newGameReducer, NewGameState } from './NewGameReducer';
+import {
+  getColors,
+  getMinGameNameLength,
+  getMaxGameNameLength,
+  getSteps,
+  getDefaultTeams,
+  getDefaultTeamColors,
+  getDefaultGameRules,
+  getStartingStep,
+} from './NewGameConstants';
+import { TeamColor, GameListItem } from '../NewGameTypes';
+import {
+  colorsAction,
+  predefinedRulesAction,
+  teamsAction,
+  customRulesAction,
+  completeStepAction,
+  setupAction,
+} from './NewGameActions';
+
+const initialState: NewGameState = {
+  setup: {
+    gameName: '',
+  },
+  teams: getDefaultTeams(),
+  teamColors: getDefaultTeamColors(),
+  rules: getDefaultGameRules(),
+  steps: {
+    completed: [],
+    active: getStartingStep(),
+  },
+  gameSubmitted: false,
+  error: false,
+};
+
+export interface UseNewGame {
+  steps: string[];
+  colorsList: TeamColor[];
+  gameSubmitted: boolean;
+  rules: GameListItem;
+  setup: {
+    gameName: string;
+  };
+  teamColors: TeamColor[];
+  teams: number;
+  error: boolean;
+  checkStep: (step: number) => boolean;
+  activeStep: number;
+  completedSteps: number[];
+  getValidationNotes: (step: number) => string;
+  onGameNameChange: (value: string) => void;
+  onTeamColorsChange: (teamColor: TeamColor) => void;
+  onTeamsChange: (newTeams: number) => void;
+  onGameRulesChange: (payload: Partial<GameListItem>) => void;
+  onPredefinedGameRuleChange: (payload: GameListItem) => void;
+  onSetStep: (step: number) => void;
+}
+
+export const useNewGame = (): UseNewGame => {
+  const badWordsFilter = new Filter();
+  const stepsList = getSteps();
+  const colorsList = getColors();
+  const [state, dispatch] = useReducer(newGameReducer, initialState);
+  const { gameSubmitted, rules, setup, teamColors, teams, steps, error } = state;
+  const { active: activeStep, completed: completedSteps } = steps;
+
+  const checkStep = useCallback(
+    (step: number): boolean => {
+      const minGameNameLength = getMinGameNameLength();
+      const maxGameNameLength = getMaxGameNameLength();
+
+      switch (step) {
+        case 0: {
+          const { gameName } = setup;
+          const wordCheck = /[*<>~]+/.test(gameName);
+          const badWords = badWordsFilter.isProfane(gameName);
+
+          return (
+            !badWords &&
+            !wordCheck &&
+            Boolean(gameName && gameName.length >= minGameNameLength && gameName.length < maxGameNameLength)
+          );
+        }
+        case 1: {
+          const {
+            winningScoreEnabled,
+            scoringSystem,
+            winningScore,
+            startingScore,
+            isMatchesBased,
+            minTeamSize,
+            maxTeamSize,
+          } = rules;
+
+          if (winningScoreEnabled && scoringSystem === 'increase' && Number(winningScore) < Number(startingScore)) {
```

Comment:
Same one as before - explicitly casting to `number`

Reply:
yes, same reason. those values could be strings